pkg/steps/template_test.go

```diff
@@ -0,0 +1,367 @@
+package steps
+
+import (
+	"testing"
+
+	coreapi "k8s.io/api/core/v1"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/pod-utils/downwardapi"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/resource"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/diff"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+
+	templateapi "github.com/openshift/api/template/v1"
+
+	"github.com/openshift/ci-tools/pkg/api"
+)
+
+func TestInjectResourcesToPod(t *testing.T) {
+	corev1Scheme := runtime.NewScheme()
+	utilruntime.Must(coreapi.AddToScheme(corev1Scheme))
+	corev1Codec := serializer.NewCodecFactory(corev1Scheme).LegacyCodec(coreapi.SchemeGroupVersion)
+
+	testCases := []struct {
+		testID    string
+		resources api.ResourceConfiguration
+		template  *templateapi.Template
+		expected  *templateapi.Template
+	}{
+		{
+			testID: "no resource requests are defined, pod with container named 'test' exists but is not changed",
+			template: &templateapi.Template{
+				TypeMeta:   meta.TypeMeta{Kind: "Template", APIVersion: "template.openshift.io/v1"},
+				ObjectMeta: meta.ObjectMeta{Name: "test-template"},
+				Objects: []runtime.RawExtension{
+					func() runtime.RawExtension {
+						pod := &coreapi.Pod{
+							TypeMeta:   meta.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+							ObjectMeta: meta.ObjectMeta{Name: "test-pod"},
+							Spec: coreapi.PodSpec{
+								Containers: []coreapi.Container{{Name: "test"}},
+							},
+						}
+						return runtime.RawExtension{
+							Raw:    []byte(runtime.EncodeOrDie(corev1Codec, pod)),
+							Object: pod.DeepCopyObject(),
+						}
+					}(),
+				},
+			},
+			expected: &templateapi.Template{
+				TypeMeta:   meta.TypeMeta{Kind: "Template", APIVersion: "template.openshift.io/v1"},
+				ObjectMeta: meta.ObjectMeta{Name: "test-template"},
+				Objects: []runtime.RawExtension{
+					func() runtime.RawExtension {
+						pod := &coreapi.Pod{
+							TypeMeta:   meta.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+							ObjectMeta: meta.ObjectMeta{Name: "test-pod"},
+							Spec: coreapi.PodSpec{
+								Containers: []coreapi.Container{{Name: "test"}},
+							},
+						}
+						return runtime.RawExtension{
+							Raw:    []byte(runtime.EncodeOrDie(corev1Codec, pod)),
+							Object: pod.DeepCopyObject(),
+						}
+					}(),
+				},
+			},
+		},
+		{
+			testID: "resource requests are defined, pod has no container named 'test' and is not changed",
+			resources: api.ResourceConfiguration{
+				"test-template": {
```

Comment:
Do we need a case for when nothing is defined for the specific template name?

Reply:
I don't think that we need to check that in these tests. This case should belong to [resourcesFor](https://github.com/openshift/ci-tools/blob/master/pkg/steps/source.go#L469) unit tests.